sql/core/src/main/scala/org/apache/spark/sql/execution/bucketing/CoalesceOrRepartitionBucketsInJoin.scala

```diff
@@ -83,23 +82,39 @@ case class CoalesceBucketsInJoin(conf: SQLConf) extends Rule[SparkPlan] {
   }
 
   def apply(plan: SparkPlan): SparkPlan = {
-    if (!conf.coalesceBucketsInJoinEnabled) {
+    if (!conf.coalesceBucketsInJoinEnabled && !conf.repartitionBucketsInJoinEnabled) {
       return plan
     }
 
+    if (conf.coalesceBucketsInJoinEnabled && conf.repartitionBucketsInJoinEnabled) {
+      throw new AnalysisException("Both 'spark.sql.bucketing.coalesceBucketsInJoin.enabled' and " +
+        "'spark.sql.bucketing.repartitionBucketsInJoin.enabled' cannot be set to true at the" +
+        "same time")
```

Comment:
For example;
```
  object BucketReadStrategyMode extends Enumeration {
    val COALESCING, REPARTITIONING, AUTOMATIC, OFF  = Value
  }

  val BUCKET_READ_STORATEGY_MODE =
    .buildConf("...")
    .version("3.1.0")
    .stringConf
    .transform(_.toUpperCase(Locale.ROOT))
    .checkValues(BucketReadStrategyMode.values.map(_.toString))
    .createWithDefault(BucketReadStrategyMode.OFF.toString)
```

Reply:
Thanks for the suggestion! I think the new config makes more sense. I renamed few, and let me know if it doesn't make sense.

Btw, do you think I can introduce `AUTOMATIC` as a follow up since this PR is sizable? Let me know if you want to see it in this PR. Thanks.